README.md

```diff
@@ -25,146 +25,125 @@ This deployment is **NOT** intended for a production environment.
 
-```
-# Set Docker Machine SoftLayer driver options
-export DRIVER_OPTS="--driver softlayer \
-  --softlayer-api-key <my_softlayer_api_key> \
-  --softlayer-user <my_softlayer_username> \
-  --softlayer-domain mydomain \
-  --softlayer-cpu 16 \
-  --softlayer-memory 65536 \
-  --softlayer-disk-size 100 \
-  --softlayer-region wdc01"
-
-# Create a machine named jupyterhub
-docker-machine create $DRIVER_OPTS jupyterhub
+This deployment uses GitHub OAuth to authenticate users.
+It requires that you create a [GitHub application](https://github.com/settings/applications/new).
+You will need to specify an OAuth callback URL in the following form:
 
 ```
+https://<myhost.mydomain>/hub/oauth_callback
+```
 
-### Connect to an existing host
-
-This example configures a "machine" on your local workstation that connects to an existing remote host at IP address `10.0.0.10`.  To do this, you must use Docker Machine's `generic` driver, and your local workstation must have a private SSH key that allows you to perform password-less login to the host.
-
-Substitute your own IP address and path to SSH key in the `DRIVER_OPTS` below.  Note that when you run the `docker-machine create` command, Docker Machine will install and configure the latest Docker Engine on the remote host (or upgrade the Docker Engine on the host if it is already installed).
+You must pass the secrets that GitHub provides for your application to JupyterHub at runtime.
+You can do this by setting the `GITHUB_CLIENT_ID`, `GITHUB_CLIENT_SECRET`,
+and `OAUTH_CALLBACK_URL` environment variables when you run the JupyterHub container,
+or you can add them to the `.env` file in the root directory of this repository.  For example,
 
 ```
-# Use the generic driver to create a Docker machine that
-# controls the Docker daemon on an existing host
-export DRIVER_OPTS="--driver generic \
-  --generic-ip-address 10.0.0.10 \
-  --generic-ssh-key /Users/jtyberg/.ssh/myhost_rsa.pem"
-
-# Create a machine named jupyterhub
-docker-machine create $DRIVER_OPTS jupyterhub
+GITHUB_CLIENT_ID=<github_client_id>
+GITHUB_CLIENT_SECRET=<github_client_secret>
+OAUTH_CALLBACK_URL=https://<myhost.mydomain>/hub/oauth_callback
 ```
 
-## Activate Docker Machine
-
-You must set specific `DOCKER_*` environment variables to tell `docker` that it should run commands against a particular machine.  The remainder of this document assumes that the machine called `jupyterhub` is active, and therefore, all `docker` commands will run on the `jupyterhub` remote host.
+**Note:** The `.env` file is a special file that Docker Compose uses to lookup environment variables.
+If you choose to place the GitHub secrets in this file,
+you should ensure that this file remains private
+(e.g., do not commit the secrets to source control).
 
-To set the `jupyterhub` machine as active:
+## Build the JupyterHub Docker image
 
-```
-eval "$(docker-machine env jupyterhub)"
-```
+Configure JupyterHub and build it into a Docker image.
```

Comment:
To configure JupyterHub and build a Docker image containing JupyterHub, you should follow these steps:

Makes it a bit clearer for new users.